pkg/querier/queryrange/stats.go

```diff
@@ -0,0 +1,55 @@
+package queryrange
+
+import (
+	"context"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+	"github.com/cortexproject/cortex/pkg/util/spanlogger"
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logql"
+	"github.com/grafana/loki/pkg/logql/stats"
+)
+
+// RecordMetrics is the function used to record metrics, overridden during tests.
+var RecordMetrics = func(status, query string, rangeType logql.QueryRangeType, stats stats.Result) {
```

Comment:
if this is only parameterized for testing purposes, I think we shouldn't export it.

Reply:
doh of course !